### PR TITLE
Support Unicode Symbols

### PR DESCRIPTION
**Pull request by ronkok**
This is the third in a series of PRs to give KaTeX the ability to recognize Unicode character input. This one supports some but not all letter-like symbols and some other symbols. All the characters in this PR will produce `textord` atoms in math mode.

The list is not comprehensive. It’s a bit eclectic. Some of these characters are important. Others were just easy to include. A few important characters are omitted because it wasn’t clear to me which is the best way to implement them. I’ll address those characters in future PRs.

### CODE REVIEW

**Comment by ronkok** [comment]
Double-struck characters cannot be done with current fonts by just setting a flag in `symbols.js`. Take ℂ, for example. In the katex-ams font, the character that looks like ℂ is not at code point U+2102 but is instead at code point U+0043. So, the `symbols.js` flag technique doesn’t work. Rather than wait for font revisions, I wrote macros.

**Comment by ronkok** [comment]
As in PR #933, the following table shows the KaTeX function to which each character is mapped and confirmation whether [unicode math](http://ctan.sharelatex.com/tex-archive/macros/latex/contrib/unicode-math/unicode-math-table.tex) and/or John Cook’s [conversion page](https://www.johndcook.com/unidata.js) map it the same way.

| **Character** | **Code point** | **Maps to**   | **unicode math** | **Cook** | **Other**       |
|:-------------:|:--------------:|:--------------|:----------------:|:--------:|:----------------|
|£  |  U+00A3  |  \pounds  |  \mathsterling  |  \sterling  |  \textsterling
|¥  |  U+00A5  |  \yen  |  \mathyen  |  ✓  |  
|ð  |  U+00F0  |  \eth  |  \matheth  |  \dh  |  
|ℂ  |  U+2102  |  \mathbb{C}  |  \BbbC  |  \BbbC  |  
|ℍ  |  U+210D  |  \mathbb{H}  |  \BbbH  |  \BbbH  |  
|ℑ  |  U+2111  |  \Im  |  ✓  |  ✓  |  \frakI
|ℓ  |  U+2113  |  \ell  |  ✓  |  ✓  |  
|ℕ  |  U+2115  |  \mathbb{N}  |  \BbbN  |  \BbbN  |  
|℘  |  U+2118  |  \wp  |  ✓  |  ✓  |  
|ℙ  |  U+2119  |  \mathbb{P}  |  \BbbP  |  \BbbP  |  
|ℚ  |  U+211A  |  \mathbb{Q}  |  \BbbQ  |  \BbbQ  |  
|ℜ  |  U+211C  |  \Re  |  ✓  |  ✓  |  \frakR
|ℝ  |  U+211D  |  \mathbb{R}  |  \BbbR  |  \BbbR  |  
|ℤ  |  U+2124  |  \mathbb{Z}  |  \BbbZ  |  \BbbZ  |  
|Ⅎ  |  U+2132  |  \Finv  |  ✓  |  ✓  |  
|ℵ  |  U+2135  |  \aleph  |  ✓  |  ✓  |  
|ℶ  |  U+2136  |  \beth  |  ✓  |  ✓  |  
|ℷ  |  U+2137  |  \gimel  |  ✓  |  ✓  |  
|ℸ  |  U+2138  |  \daleth  |  ✓  |  ✓  |  
|⅁  |  U+2141  |  \Game  |  ✓  |  ✓  |  
|∀  |  U+2200  |  \forall  |  ✓  |  ✓  |  
|∁  |  U+2201  |  \complement  |  ✓  |  ✓  |  
|∂  |  U+2202  |  \partial  |  ✓  |  ✓  |  
|∃  |  U+2203  |  \exists  |  ✓  |  ✓  |  
|∇  |  U+2207  |  \nabla  |  ✓  |  ✓  |  \del
|∞  |  U+221E  |  \infty  |  ✓  |  ✓  |  
|∠  |  U+2220  |  \angle  |  ✓  |  ✓  |  
|∡  |  U+2221  |  \measuredangle  |  ✓  |  ✓  |  
|∢  |  U+2222  |  \sphericalangle  |  ✓  |  \sphericalangle \ltrpar  |  
|♠  |  U+2660  |  \spadesuit  |  ✓  |  ✓  |  
|♡  |  U+2661  |  \heartsuit  |  ✓  |  ✓  |  
|♢  |  U+2662  |  \diamondsuit  |  ✓  |  ✓  |  
|♣  |  U+2663  |  \clubsuit  |  ✓  |  ✓  |  
|♭  |  U+266D  |  \flat  |  ✓  |  ✓  |  
|♮  |  U+266E  |  \natural  |  ✓  |  ✓  |  
|♯  |  U+266F  |  \sharp  |  ✓  |  ✓  |  
|✓  |  U+2713  |  \checkmark  |  ✓  |  ✓  |

**Comment by kevinbarabash** [comment]
> Rather than wait for font revisions, I wrote macros.

I think that's sensible.  I think revising the fonts would introduce other complications as it relates to handling font commands such as `\mathbb`, etc.

**Comment by ronkok** [comment]
Let's hold off on this until after PR #973 lands. I'll revise this PR accordingly.

**Comment by ronkok** [comment]
The most recent commit reverts a RegEx in `MacroExpander.js`. So the code in this PR is now dependent on PR #973.

**Comment by ronkok** [comment]
Now that PR #973 has landed, this should be good to go.

File: src/macros.js [hunk 1]
```diff
@@ -48,6 +48,15 @@ defineMacro("\\egroup", "}");
...
 defineMacro("\\endgroup", "}");
 
+// Unicode double-struck letters
+defineMacro("\u2102", "\\mathbb{C}");
+defineMacro("\u210D", "\\mathbb{H}");
+defineMacro("\u2115", "\\mathbb{N}");
+defineMacro("\u2119", "\\mathbb{P}");
+defineMacro("\u211A", "\\mathbb{Q}");
+defineMacro("\u211D", "\\mathbb{R}");
+defineMacro("\u2124", "\\mathbb{Z}");
```
Comment (kevinbarabash) [inline src/macros.js]: nice!

**Pull request merged**